### ISSUE TITLE
Immutability / Censorship Resistance

### ISSUE THREAD

**Issue by auryn-macmillan** [auryn-macmillan/DAOForum#6]
**How to ensure censorship resistance, long term resilience, & immutability?**

**Comment by ghost** [comment]
I touched on this a bit in the moderation issue, so check that out, but elected rotating moderation teams could mitigate some of this. 

The only real threat in terms of censorship, at least in my mind, would come from mods. I could see members who are trusted by the community being paid to spout certain opinions too, but that's more manipulation than censorship - and I don't know how to go about solving that one. 

I think this a big problem that needs to be prepared for though. The slock.it DAO is going to raise 10$ million+, so a forum that hosts that fund's governance (or any other DAOs in the future) is going to be ripe for attack and manipulation from specific parties in a way that traditional forums just aren't.

**Comment by auryn-macmillan** [comment]
The moderation side of censorship is only one side of the proverbial coin. The other thing to consider is censorship from outside actors: government, DDoS attacks, etc. How can we build the system to be resilient to censorship from large scale attacks? It could be vitally important for the forum to be active in the lead up to large votes, and there could be some very real financial incentives for some actors to hinder communications of the DAO participants.

**Comment by CaptainSkywave** [comment]
yep - security and privacy are the 2 important ones.

**Comment by vmichalik** [comment]
I think there needs to be some sort of prioritization- I don't think DDoS attacks are that relevant in the short term. In the longer term companies like CloudFlare offer DDoS protection. I think we could also store hashes of threads in the blockchain to maintain historical integrity.

**Comment by auryn-macmillan** [comment]
Yeah I agree, DDoS attacks probably aren't a concern in the short term. But long term data integrity should be considered from day one. Perhaps having redundant copies of the the forum stored on services like stoj or ipfs would be a good solution.

**Comment by ghost** [comment]
The copies should be stored using a decentralized solution, that's for certain. As for the exact technology to use (re: storj vs ipfs), I don't know enough about the differences between the two to weigh in one way or the other.

**Comment by matrixator** [comment]
I read that there will be swarm in Ethereum (similar to storj and maidsafe).

**Comment by auryn-macmillan** [comment]
I think swarm would be an ideal solution, but I doubt it will be ready by the time we're looking to implement. So we may need to use something else, at least in the interim.
